packages/api/amplify_api/lib/amplify_api.dart

```diff
@@ -36,13 +37,8 @@ abstract class AmplifyAPI extends APIPluginInterface {
     List<APIAuthProvider> authProviders = const [],
     ModelProviderInterface? modelProvider,
   }) {
-    if (zIsWeb || Platform.isWindows || Platform.isMacOS || Platform.isLinux) {
-      throw UnsupportedError('This platform is not supported yet');
-    }
-    return AmplifyAPIMethodChannel(
-      authProviders: authProviders,
-      modelProvider: modelProvider,
-    );
+    return AmplifyAPIDart(
```

Comment:
suggestion: While I think this is needed to test the code here and this would eventually be the final state once API dart finished, I think this PR should not have this change committed as we still need the method channel implementation for ios/android until dart implementation more mature.